tests/test_package_initialization.py

```diff
@@ -0,0 +1,87 @@
+import os
+import json
+
+import pytest
+
+from ethpm import Package
+
+from ethpm.exceptions import ValidationError
+
+
+def test_init_from_valid_lockfile_data():
+    minimal_lockfile = {
+      "package_name": "foo",
+      "lockfile_version": "1",
+      "version": "1.0.0",
+    }
+
+    Package(minimal_lockfile)
+
+
+def test_init_from_invalid_lockfile_data():
+    with pytest.raises(ValidationError):
+      Package({})
+
+
+def test_init_from_invalid_argument_type():
+    with pytest.raises(TypeError):
+      Package("not a lockfile")
+
+
+def test_from_file_fails_with_missing_filepath(tmpdir):
+    path = os.path.join(
+      str(tmpdir.mkdir('invalid')),
+      'lockfile.json',
+    )
+
+    assert not os.path.exists(path)
+    with pytest.raises(FileNotFoundError):
+        Package.from_file(path)
+
+
+@pytest.fixture(params=[True, False])
+def non_json_lockfile(tmpdir, request):
+    file = tmpdir.mkdir('invalid').join('lockfile.json')
+    file.write('This is invalid json')
+    if request.param:
+        yield str(file)
+    else:
+        with open(str(file)) as file_obj:
+            yield file_obj
```

Comment:
Here is a pattern for making this generic.

```python

def _generate_fixture_param(tmpfile, as_file_path, file_content):
    temp_lockfile = tmpdir.mkdir('invalid').join('lockfile.json')
    temp_lockfile.write(file_content)
    if as_file_path:
        return str(temp_lockfile)
    else:
        with open(str(file)) as file_obj:
            return file_obj


@pytest.fixture(params=[True, False])
def the_first_fixture(tmpfile, request):
    return _generate_fixture_param(tmpfile, request.param, 'This is invalid json')

@pytest.fixture(params=[True, False])
def the_second_fixture(tmpfile, request):
    return _generate_fixture_param(tmpfile, request.param, '{"something": "else"}')
```

Reply:
Awesome. I'm hoping to get some more done here this afternoon.

Reply:
@pipermerriam Did you mean `tmpdir` where you wrote `tmpfile`? I initially read it that way but now I'm hitting some bugs, so I want to be sure they're not coming from a bad assumption.

Reply:
yes, `tmpdir`, my bad.

Reply:
Awesome. I think I figured out what's wrong, hoping to get this ready for review this evening.